packages/firestore/test/unit/specs/existence_filter_spec.test.ts

```diff
@@ -49,6 +49,29 @@ describeSpec('Existence Filters:', [], () => {
       .expectEvents(query, { added: [doc1] });
   });
 
+  specTest('Existence filter ignored with pending target', [], () => {
+    const query = Query.atPath(path('collection'));
+    const doc1 = doc('collection/1', 2000, { v: 2 });
+    return (
+      spec()
+        .withGCEnabled(false)
+        .userListens(query)
+        .watchAcksFull(query, 1000, doc1)
+        .expectEvents(query, { added: [doc1] })
+        .userUnlistens(query)
+        .watchRemoves(query)
+        .userListens(query, 'resume-token-1000')
+        .expectEvents(query, { added: [doc1], fromCache: true })
+        // The empty existence filter is ignored since Watch hasn't ACKed the
+        // target
+        .watchFilters([query])
```

Comment:
I don't think watch would send an existence filter here since it already removed the target. Perhaps the `watchRemoves(query)` should be moved down below this?